src/orange/detail/UnitInserter.cc

```diff
@@ -174,6 +180,58 @@ BoundingBoxBumper<fast_real_type, real_type> make_bumper(Tolerance<> const& tol)
     return BoundingBoxBumper<fast_real_type, real_type>(std::move(bbox_tol));
 }
 
+struct ForceMax
+{
+    size_type faces = std::numeric_limits<size_type>::max();
+    size_type intersections = std::numeric_limits<size_type>::max();
+};
+
+static char const mfi_hack_envname[] = "ORANGE_MAX_FACE_INTERSECT";
+
+//---------------------------------------------------------------------------//
+/*!
+ * Force maximum faces/intersections.
+ *
+ * This is if we know the "automatic" value is wrong, specifically if all
+ * complicated/background cells are unreachable.
+ *
+ * See https://github.com/celeritas-project/celeritas/issues/1334 .
+ */
+ForceMax const& forced_scalar_max()
+{
+    static ForceMax const result = [] {
+        std::string mfi = celeritas::getenv(mfi_hack_envname);
+        if (mfi.empty())
+        {
+            return ForceMax{};
+        }
+        CELER_LOG(warning)
+            << "Using a temporary, unsupported, and dangerous hack to "
+               "override maximum faces and intersections in ORANGE: "
+            << mfi_hack_envname << "='" << mfi << "'";
+
+        ForceMax result;
+        static std::regex const mfi_regex{R"re(^(\d+),(\d+)$)re"};
```

Comment:
In the context of this being a presumably temporary hack I don't know if this matters, but it the regex validation is complex enough to get its own function in anonymous namespace.

Reply:
Yeah this option shouldn't live long (the "cheat" itself isn't tested anywhere except by me manually for our regression problems). The regex `^(\d+),(\d+)$` isn't that complicated (whole text, capture two nonnegative integers separated by a comma) but perhaps it's harder to read because of the literal `R"re()re"` string.